fitsio/fitslib.py

```diff
@@ -1720,6 +1723,11 @@ def write_column(self, column, data, **keys):
             # some logic
             data_send = array_to_native(data, inplace=False)
 
+        if IS_PY3 and data_send.dtype.char == 'U':
+            # for python3, we convert unicode to ascii
+            # this will error if the character is not in ascii
+            data_send = data_send.astype('S', copy=False)
```

Comment:
This block should not be needed since this was done in `array_to_native`. However, for whatever reason, the tests would not pass without it. IDK.

Reply:
I fixed this. Had to do with an edge case for in place array conversion.